framework/sandstone_context_dump.cpp

```diff
@@ -334,6 +337,27 @@ void dump_gprs(std::string &f, SandstoneMachineContext mc)
 }
 #endif
 
+static void print_egprs(std::string &f, const Fxsave *state)
+{
+    int offset = xsave_offset(XSave_ApxState);
+    if (offset + sizeof(ApxState) <= Fxsave::size)
+        return;
+
+    auto base = reinterpret_cast<const uint8_t *>(state);
+    auto egprs = reinterpret_cast<const ApxState *>(base + offset);
+    char regname[] = "r16";
+    for (int64_t value : *egprs) {
+        print_gpr(f, regname, value);
+
+        // increment the register name
+        if (regname[2]++ == '9') {
```

Comment:
Why do this if it's a fix arch register set? Let's just have an array of names (`static const *char egpr_names[16] = { "r16", ... };`). Or even counter and `sprintf()` to `char[3]`.

Reply:
That would be far more expensive than this, requiring static memory for the register names and an array, or for the `sprintf` call, the latter of which would do exactly what we're doing here.